libs/auth/src/openssl.ts

```diff
@@ -186,68 +211,254 @@ export async function verifySignature({
+   */
+  certKeyInput:
+    | { type: 'private'; key: FileKey }
+    | { type: 'public'; key: InlineKey };
+  certSubject: string;
+  certType?: CertType;
+  expiryInDays: number;
+  signingCertAuthorityCertPath: string;
+  signingPrivateKey: FileKey | TpmKey;
+}
+
+const CreateCertInputSchema: z.ZodSchema<CreateCertInput> = z.object({
+  certKeyInput: z.union([
+    z.object({ type: z.literal('private'), key: FileKeySchema }),
+    z.object({ type: z.literal('public'), key: InlineKeySchema }),
+  ]),
+  certSubject: z.string(),
+  certType: z
+    .union([z.literal('cert_authority_cert'), z.literal('standard_cert')])
+    .optional(),
+  expiryInDays: z.number(),
+  signingCertAuthorityCertPath: z.string(),
+  signingPrivateKey: z.union([FileKeySchema, TpmKeySchema]),
+});
+
+/**
+ * Parses a JSON string as a CreateCertInput, throwing an error if parsing fails
+ */
+export function parseCreateCertInput(value: string): CreateCertInput {
+  return unsafeParse(CreateCertInputSchema, JSON.parse(value));
+}
+
+/**
+ * A helper for createCert. Creates a cert, combining cert signing request creation and cert
+ * creation into a single function. The outputted cert will be in PEM format.
+ */
+export async function createCertHelper({
+  certKeyInput,
+  certSubject,
+  certType = 'standard_cert',
+  expiryInDays,
+  signingCertAuthorityCertPath,
+  signingPrivateKey,
+}: CreateCertInput): Promise<Buffer> {
+  const isPrivateKeyOfPublicKeyToSignUnavailable =
+    certKeyInput.type === 'public';
+
+  let certKey: FileKey | InlineKey;
+  if (isPrivateKeyOfPublicKeyToSignUnavailable) {
+    const throwawayPrivateKey = await openssl([
+      'ecparam',
+      '-genkey',
+      '-name',
+      'prime256v1',
+      '-noout',
+    ]);
+    certKey = {
+      source: 'inline',
+      content: throwawayPrivateKey.toString('utf-8'),
+    };
+  } else {
+    certKey = certKeyInput.key;
+  }
+  const certSigningRequest = await createCertSigningRequest({
+    certKey,
+    certSubject,
+  });
+
+  const certPublicKeyOverride = isPrivateKeyOfPublicKeyToSignUnavailable
+    ? Buffer.from(certKeyInput.key.content, 'utf-8')
+    : undefined;
+  const cert = await createCertFromCertSigningRequest({
+    certPublicKeyOverride,
+    certSigningRequest,
+    certType,
+    expiryInDays,
+    signingCertAuthorityCertPath,
+    signingPrivateKey,
+  });
+
+  return cert;
+}
+
+/**
+ * Creates a cert, combining cert signing request creation and cert creation into a single
+ * function. The outputted cert will be in PEM format.
+ *
+ * Uses a slightly roundabout control flow for permissions purposes:
+ *
+ * createCert() --> ../src/create-cert --> createCertHelper()
+ *
+ * When using TPM keys in production, we need sudo access. Creating the intermediate create-cert
+ * script allows us to grant password-less sudo access to the relevant system users for just that
+ * operation, instead of having to grant password-less sudo access more globally.
```

Comment:
I'll be updating the [sudoers file in vxsuite-complete-system](https://github.com/votingworks/vxsuite-complete-system/blob/c77994a22449046ddcb1d0e18dcafe7112850215/config/sudoers#L22-L41) to accomplish this granular access control